src/settings.cpp

```diff
@@ -37,17 +37,155 @@ static Settings main_settings;
 Settings *g_settings = &main_settings;
 std::string g_settings_path;
 
-Settings::~Settings()
+static std::map<std::string, MTSetting> mtsettings_names; //TODO change to unordered_map
+
+static MTSetting get_mtsetting_for_name(const std::string &name)
+{
+	if (mtsettings_names.empty()) {
+#define MTCONF_SET_EXT(enumname, lcase, default)\
+	mtsettings_names[lcase] = CNF_##enumname;
+#include "settingdef.h"
+#undef MTCONF_SET_EXT
+	}
+	std::map<std::string, MTSetting>::iterator it;
+	it = mtsettings_names.find(name);
+	if (it != mtsettings_names.end()) {
+		return it->second;
+	} else {
+		return CNF_COUNT;
+	}
+}
+
+
+static std::string get_name_of_mtsetting(MTSetting setting)
+{
+	switch(setting) {
+#define MTCONF_SET_EXT(enumname, lcase, default)\
```

Comment:
I think it's better to use a static const struct MTSetting [MAX_SETTINGS] { SETTING_ENUM_ID_1, "SETTING_ENUM_ID_1" } and lookup directly at the right ID instead of doing many and many cases at search call